src/discretization/1_discretization.cpp

```diff
@@ -0,0 +1,65 @@
+
+#include "0_staggered_grid.h"
+
+// Contructor
+Discretization::Discretization(std::array<int,2> nCells, std::array<double,2> meshWidth) :
+    StaggeredGrid(nCells, meshWidth)
+{
+}
+
+
+//! compute the 2nd derivative ∂^2 u / ∂x^2 at right edge
+double computeD2uDx2(int i, int j) const
+{
+    const double hx = this->meshWidth_[0] ; // mesh width in x direction
+
+    // return 2nd derivative ddu/dxx at right edge of cell i,j
+    return 1./(hx*hx) * (u(i-1,j) - 2.*u(i,j) + u(i-1,j));
```

Comment:
The members of the StaggeredGrid-instances are FieldVariable with an underscore behind the variable
```suggestion
    return 1./(hx*hx) * (u_(i-1,j) - 2.*u_(i,j) + u_(i-1,j));
```